core/config/configtest/config.go

```diff
@@ -38,15 +43,55 @@ func dirExists(path string) bool {
 // maintained with the source tree. This should only be used in a
 // test/development context.
 func GetDevConfigDir() string {
-	gopath := os.Getenv("GOPATH")
+	path, err := gomodDevConfigDir()
+	if err != nil {
+		path, err = gopathDevConfigDir()
+		if err != nil {
+			panic(err)
+		}
+	}
+	return path
```

Comment:
The barest of nits. This reads better straight line with no indentation. I don't think the single exit point helps in a function this short. I think it's clearer in terms of, use gomod first, use gopath second.

```
	path, err := gomodDevConfigDir()
	if err == nil {
		return path
	}
	path, err = gopathDevConfigDir()
	if err == nil {
		return path
	}
	panic(err)
```
It can even be collapsed with the assignable-if
```
	var err error
	if path, err := gomodDevConfigDir(); err == nil {
		return path
	}
	if path, err := gopathDevConfigDir(); err == nil {
		return path
	}
	panic(err)

Reply:
I thought about what you suggested above. But, we need three `return path` statement and especially the last `return path` which comes after `panic(err)` is **a dead code**. IMO, the current structure looks good and no changes required.

Reply:
@MHBauer If the `panic(err)` can be avoided, what you suggested is applicable and makes the code more readable.